lib/policyEvaluator/RequestContext.js

```diff
@@ -0,0 +1,400 @@
+ * @return {RequestContext} a RequestContext instance
+ */
+
+class RequestContext {
+    constructor(headers, query, bucketName, objectName,
+        requesterIp, isEncrypted, apiMethod,
+        awsService, locationConstraint, requesterInfo,
+        signatureVersion, authType, signatureAge) {
+        this._headers = headers;
+        this._query = query;
+        this._requesterIp = requesterIp;
+        this._sslEnabled = isEncrypted;
+        this._apiMethod = apiMethod;
+        this._awsService = awsService;
+        this._bucket = bucketName;
+        this._object = objectName;
+        this._locationConstraint = locationConstraint;
+        // Not implemented
+        this._multiFactorAuthPresent = undefined;
+        // Not implemented
+        this._multiFactorAuthAge = undefined;
+        // Not implemented
+        this._tokenIssueTime = undefined;
+
+        // Remainder not set when originally instantiated
+        // (unless if instantiated from deSerialize)
+        this._requesterInfo = requesterInfo;
+        // See http://docs.aws.amazon.com/AmazonS3/latest/
+        // API/bucket-policy-s3-sigv4-conditions.html
+        this._signatureVersion = signatureVersion;
+        this._authType = authType;
+        this._signatureAge = signatureAge;
+
+        return this;
+    }
+
+    /**
+    * Serialize the object
+    * @return {string} - stringified object
+    */
+    serialize() {
+        const requestInfo = {
+            apiMethod: this._apiMethod,
+            headers: this._headers,
+            query: this._query,
+            requersterInfo: this._requesterInfo,
+            requesterIp: this._requesterIp,
+            sslEnabled: this._sslEnabled,
+            awsService: this._awsService,
+            bucket: this._bucket,
+            object: this._object,
+            multiFactorAuthPresent: this._multiFactorAuthPresent,
+            multiFactorAuthAge: this._multiFactorAuthAge,
+            signatureVersion: this._signatureVersion,
+            authType: this._authType,
+            signatureAge: this._signatureAge,
+            locationConstraint: this._locationConstraint,
+            tokenIssueTime: this._tokenIssueTime,
+        };
+        return JSON.stringify(requestInfo);
+    }
+
+    /**
+     * deSerialize the JSON string
+     * @param {string} stringRequest - the stringified requestContext
+     * @return {object} - parsed string
+     */
+    static deSerialize(stringRequest) {
+        const obj = JSON.parse(stringRequest);
+        return new RequestContext(obj.headers, obj.query, obj.bucket,
+            obj.object, obj.requesterIp, obj.sslEnabled, obj.apiMethod,
+            obj.awsService, obj.locationConstraint, obj.requesterInfo,
+            obj.signatureVersion, obj.authType, obj.signatureAge);
+    }
+
+    /**
+    * Get the request action
+    * @return {string} action
+    */
+    getAction() {
+        return _findAction(this._awsService, this._apiMethod);
```

Comment:
For performance consideration, create another variable to store the result of the _findAction in the constructor :)
nb: same for getRessource()